swaybar/bar.c

```diff
@@ -262,9 +305,20 @@ void bar_setup(struct swaybar *bar,
 	wl_registry_add_listener(registry, &registry_listener, bar);
 	wl_display_roundtrip(bar->display);
 	assert(bar->compositor && bar->layer_shell && bar->shm);
+	wl_display_roundtrip(bar->display);
+
 	struct swaybar_pointer *pointer = &bar->pointer;
 
-	assert(pointer->cursor_theme = wl_cursor_theme_load(NULL, 16, bar->shm));
+	int max_scale = 1;
+	struct swaybar_output *output;
+	wl_list_for_each(output, &bar->outputs, link) {
+		if (output->scale > max_scale) {
+			max_scale = output->scale;
+		}
+	}
+
+	assert(pointer->cursor_theme = wl_cursor_theme_load(
```

Comment:
You shouldn't put a side effect in an assertion.

Reply:
I do this all the time, it's all over sway et al. This is fine.

Reply:
Nevermind duh I see why

Reply:
Pushed fix to this file but will file a follow-up ticket to go fix my fuck-ups elsewhere